File: 프로그래머스/0/120913. 잘라서 배열로 저장하기/잘라서 배열로 저장하기.java
class Solution {
    public String[] solution(String my_str, int n) {
        int len = my_str.length() / n + (my_str.length() % n != 0 ? 1 : 0);
        String[] answer = new String[len];
        
        for(int i = 0; i < len; i++) {
            int index = my_str.length() >= n ? n : my_str.length();
            answer[i] = my_str.substring(0,index);
            my_str = my_str.substring(index);
        }
        
        return answer;
    }
}

File: 프로그래머스/0/120913. 잘라서 배열로 저장하기/잘라서 배열로 저장하기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

char** solution(const char* my_str, int n) {
    int len = (strlen(my_str) + n - 1) / n;
    int idx = 0;
    char** answer = (char**)malloc(len * sizeof(char*));
    
    for(int i = 0; i < len; i++){
        answer[i] = (char*)malloc(n * sizeof(char));
        idx = 0;
        
        while(*my_str != '\0' && idx < n){
            answer[i][idx++] = *my_str;
            my_str++;
        }
        answer[i][idx] = '\0';
    }
    
    return answer;
}